### PR TITLE
Fix up socket tests.

### LINKED ISSUE #17505: Test fails on linux mint
### Description

There were 2 failures:

1) Cake\Test\TestCase\Network\SocketTest::testEnableCryptoSocketExceptionNoSsl
Failed asserting that null is an instance of class Exception.
/vendor/cakephp/cakephp/tests/TestCase/Network/SocketTest.php:292

2) Cake\Test\TestCase\Network\SocketTest::testEnableCryptoSocketExceptionNoTls
Failed asserting that null is an instance of class Exception.
/vendor/cakephp/cakephp/tests/TestCase/Network/SocketTest.php:321

When running cake5 tests locally on Linux Mint, so apparently the "null" return is not only Windows specific
```php
        $this->assertNotNull($e);
        // Windows doesn't return the previous exception for some reason
        if (DS !== '\\') {
            $this->assertInstanceOf('Exception', $e->getPrevious());
        }
```

### CakePHP Version

5.x

### PHP Version

8.2/8.3

### PR DESCRIPTION
**Pull request by dereuromark**
Resolves https://github.com/cakephp/cakephp/issues/17505

and pin down PHPUnit for now.

### CODE REVIEW

File: tests/TestCase/Network/SocketTest.php [hunk 1]
```diff
@@ -287,10 +287,6 @@ public function testEnableCryptoSocketExceptionNoSsl(): void
         }
 
         $this->assertNotNull($e);
```
Comment (othercorey) [inline tests/TestCase/Network/SocketTest.php:289]: This looks like we're relying on a PHP quirk. We should save the exception in a different variable.

Comment (dereuromark) [reply tests/TestCase/Network/SocketTest.php:289]: Sure, this is just a quick fix for green CI for now.

**Comment by othercorey** [comment]
Looking at the Socket code, I don't understand how it can not be set. Will let @markstory  review.

**Comment by LordSimal** [comment]
Looking at https://github.com/cakephp/cakephp/pull/17509 I'd rather adjust the range to `^10.1.0 <=10.5.3` so we can keep BC.

The new behaviour can be changed in `5.next`

**Comment by dereuromark** [comment]
Sure, we can do that in this PR or in another one and I rebase this one accordingly and revert the phpunit change.

**Comment by dereuromark** [comment]
I reverted the composer.json change.

**Pull request merged**